fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/DimensionSearchServlet.java

```diff
@@ -0,0 +1,198 @@
+import com.yahoo.bard.webservice.exception.MetadataExceptionHandler;
+import com.yahoo.bard.webservice.logging.RequestLog;
+import com.yahoo.bard.webservice.logging.blocks.DimensionRequest;
+import com.yahoo.bard.webservice.table.LogicalTableDictionary;
+import com.yahoo.bard.webservice.util.Pagination;
+import com.yahoo.bard.webservice.util.StreamUtils;
+import com.yahoo.bard.webservice.web.RequestMapper;
+import com.yahoo.bard.webservice.web.ResponseFormatResolver;
+import com.yahoo.bard.webservice.web.apirequest.ApiRequestImpl;
+import com.yahoo.bard.webservice.web.apirequest.DimensionsApiRequest;
+import com.yahoo.bard.webservice.web.apirequest.DimensionsApiRequestImpl;
+import com.yahoo.bard.webservice.web.apirequest.ResponsePaginator;
+import com.yahoo.bard.webservice.web.util.PaginationParameters;
+import com.yahoo.bard.webservice.data.dimension.SearchQuerySearchProvider;
+
+import com.codahale.metrics.annotation.Timed;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+/**
+ * Resource code for dimension resource endpoints.
+ */
+@Singleton
+@Path("dimensions/{dimensionName}/search")
+public class DimensionSearchServlet extends EndpointServlet {
+    private static final Logger LOG = LoggerFactory.getLogger(DimensionSearchServlet.class);
+
+    private final DimensionDictionary dimensionDictionary;
+    private final RequestMapper requestMapper;
+    private final ResponseFormatResolver formatResolver;
+    private final MetadataExceptionHandler exceptionHandler;
+
+    /**
+     * Constructor.
+     *
+     * @param dimensionDictionary  All dimensions
+     * @param logicalTableDictionary  All logical tables
+     * @param requestMapper  Mapper to change the API request if needed
+     * @param objectMappers  JSON tools
+     * @param formatResolver  The formatResolver for determining correct response format
+     * @param exceptionHandler  Injection point for handling response exceptions
+     */
+    @Inject
+    public DimensionSearchServlet(
+            DimensionDictionary dimensionDictionary,
+            LogicalTableDictionary logicalTableDictionary,
+            @Named(DimensionsApiRequest.REQUEST_MAPPER_NAMESPACE) RequestMapper requestMapper,
+            ObjectMappersSuite objectMappers,
+            ResponseFormatResolver formatResolver,
+            @Named(DimensionsApiRequest.EXCEPTION_HANDLER_NAMESPACE) MetadataExceptionHandler exceptionHandler
```

Comment:
Do we want to give this servlet its own namespace for request mappers and exception handling, since it is
its own servlet?

Reply:
I'd prefer not, this being a simple add-on to the dimension servlet namespace.  But I could be persuaded.

Reply:
I don't particularly care. And if it's intended to be used as an add-on the dimension servlet, as opposed to a servlet in its own right with its own rules, then having it use the same namespace makes sense.

Basically, "This is coupling these two servlets together. Do we want them coupled together?" If the answer is 'yes,' then keep it as is.

Reply:
Will add documentation